Added CUAD dataset

bhavitvyamalik opened the pull request:
Dataset link : https://github.com/TheAtticusProject/cuad/

Working on README.md currently.

Closes #2084 and [#1](https://github.com/TheAtticusProject/cuad/issues/1).

lhoestq commented on datasets/cuad/README.md:
```diff
@@ -0,0 +1,228 @@
+
+Contract review costs also affect consumers. Since contract review costs are so prohibitive, contract review is not often performed outside corporate transactions. Small companies and individuals consequently often sign contracts without even reading them, which can result in predatory behavior that harms consumers. Automating contract review by openly releasing high-quality data and fine-tuned m ...
+
+To reduce the disparate societal costs of contract review, and to study how well NLP models generalize to specialized domains, we introduce a new large-scale dataset for contract review. As part of The Atticus Project, a non-profit organization of legal experts, we introduce CUAD, the Contract Understanding Atticus Dataset. This dataset was created with a year-long effort pushed forward by dozens ...
```
```suggestion
To reduce the disparate societal costs of contract review, and to study how well NLP models generalize to specialized domains, the authors introduce a new large-scale dataset for contract review. As part of The Atticus Project, a non-profit organization of legal experts, CUAD is introduced, the Contract Understanding Atticus Dataset. This dataset was created with a year-long effort pushed forward by dozens of law student annotators, lawyers, and machine learning researchers. The dataset includes more than 500 contracts and more than 13,000 expert annotations that span 41 label categories. For each of 41 different labels, models must learn to highlight the portions of a contract most salient to that label. This makes the task a matter of finding needles in a haystack.
```
Maybe we can try to avoid the use of "we" everywhere to refer to the authors, since we=HF community

bhavitvyamalik replied:
Few touchups were in to-do for the README. Thanks!

lhoestq commented on datasets/cuad/README.md:
```diff
@@ -0,0 +1,228 @@
+
+#### Initial Data Collection and Normalization
+
+The CUAD includes commercial contracts selected from 25 different types of contracts based on the contract names as shown below. Within each type, we randomly selected contracts based on the names of the filing companies across the alphabet.
```
```suggestion
The CUAD includes commercial contracts selected from 25 different types of contracts based on the contract names as shown below. Within each type, the creators randomly selected contracts based on the names of the filing companies across the alphabet.
```

lhoestq commented on datasets/cuad/README.md:
```diff
@@ -0,0 +1,228 @@
+
+#### Annotation process
+
+Our labeling process included multiple steps to ensure accuracy:
```
```suggestion
The labeling process included multiple steps to ensure accuracy:
```

lhoestq commented on datasets/cuad/README.md:
```diff
@@ -0,0 +1,228 @@
+
+### Personal and Sensitive Information
+
+Some clauses in the files are redacted because the party submitting these contracts redacted them to protect confidentiality. Such redaction may show up as asterisks (***) or underscores (___) or blank spaces. The dataset and the answers reflect such redactions. For example, the answer for “January __ 2020” would be “1/[]/2020”).
```
```suggestion
Some clauses in the files are redacted because the party submitting these contracts redacted them to protect confidentiality. Such redaction may show up as asterisks (\*\*\*) or underscores (\_\_\_) or blank spaces. The dataset and the answers reflect such redactions. For example, the answer for “January \_\_ 2020” would be “1/[]/2020”).
```
escape stars and underscores

lhoestq reviewed (commented):
Awesome thank you !
This looks very good :)

I left a few comments for the dataset card

Also it looks like the dummy data are quite big (200KB+), do you think you could reduce their size ? For example if you could include only 1-2 examples instead of 5 that's be good

bhavitvyamalik commented on datasets/cuad/README.md:
```diff
@@ -0,0 +1,228 @@
+---
+YAML tags:
+- copy-paste the tags obtained with the tagging app: https://github.com/huggingface/datasets-tagging
```
Just started with that!

bhavitvyamalik commented:
1) Changed the language in a few places apart from those you mentioned in README
2) Reduced the size of dummy data folder by removing all other entries except the first
3) Updated YAML tags by using to the past version of `datasets-tagging` app. Will update the quick fix on that repository too in a while

lhoestq commented on datasets/cuad/README.md:
```diff
@@ -0,0 +1,251 @@
+task_categories:
+- question-answering
+task_ids:
+- closed-domain-qa
```
```suggestion
- closed-domain-qa
- extractive-qa
```

lhoestq reviewed (approved):
Thanks ! Looks all good :)

I just added the `extractive-qa` tag

MohammedRakib commented:
@bhavitvyamalik  Thanks for adding the dataset on huggingface! Can you please add a metric also for the dataset using the squad_v2 metric file?

bhavitvyamalik commented:
@MohammedRakib you can check [#2257](https://github.com/huggingface/datasets/pull/2257)
